File: server/flask_server/app/routes/news_route.py
from flask import Blueprint, jsonify
from app.controller.news_controller import get_news, get_news_by_url

news_route = Blueprint('news_route', __name__)


@news_route.route('/news', methods=['GET'])
def get_news_route():
    news_titles, status_code = get_news()
    return jsonify(news_titles), status_code

@news_route.route('/news/<path:url>', methods=['GET'])
def get_news_by_url_route(url):
    news_article, status_code = get_news_by_url(url)
    return jsonify(news_article), status_code

File: flask_server_onnode/src/routes/news_route.js
const express = require("express");
const { getNews, getNewsByUrl } = require("../controller/news_controller");

const router = express.Router();

router.get("/news", async (req, res, next) => {
  try {
    const { newsTitles, statusCode } = await getNews();
    res.status(statusCode).json(newsTitles);
  } catch (error) {
    next(error);
  }
});

router.get("/news/:url", async (req, res, next) => {
  try {
    const { newsArticle, statusCode } = await getNewsByUrl(req.params.url);
    res.status(statusCode).json(newsArticle);
  } catch (error) {
    next(error);
  }
});

module.exports = router;